tests/e2e_test.go

```diff
@@ -0,0 +1,293 @@
+package tests
```

Comment:
Would it make sense to use a build tag here? This only needs to be built during E2E tests.